src/main/java/io/leonis/zosma/ipc/serialization/protobuf/SSLVisionFunction.java

```diff
@@ -29,16 +29,17 @@ public VisionPacket apply(final WrapperPacket wrapperPacket) throws Exception {
         .join(Mx.first(WrapperPacket::getGeometry)
           .add(new GeometryFunction()))
         .join(Mx.first(WrapperPacket::getDetection)
-          .add(new DetectionFrameFunction()))
+          .add(new BallsSelector()))
+//          .add(new PlayersSelector())) // TODO I need the blueTeamAllegiance (and thus referee)
         .demux(VisionPacket::new);
   }
 
-  @AllArgsConstructor
+  @Value @AllArgsConstructor
   public final static class VisionPacket implements Temporal {
     @Delegate
-    private final GeometryFrame geometryContainer;
-    @Delegate
-    private final DetectionFrame detectionContainer;
+    private final Field field;
+    private final Set<Ball> balls;
+//    private final Set<Player> players;
```

Comment:
??

Reply:
See the TODO above; this thing still needs `blueTeamAllegiance` or something to sort the players according to allegiance. but i'm not happy with this class in general so i might remove it entirely